### PR TITLE
feat(snapshot): delete snapshot grpc call

### PR DESCRIPTION
**Pull request by dsharma-dc**

### CODE REVIEW

File: control-plane/agents/src/bin/core/controller/io_engine/mod.rs [hunk 1]
```diff
@@ -156,6 +156,9 @@ pub(crate) trait ReplicaSnapshotApi {
         &self,
         request: &CreateReplicaSnapshot,
     ) -> Result<CreateReplicaSnapshotResp, SvcError>;
+
+    /// Remove a snapshot using the incoming id.
+    async fn delete_repl_snapshot(&self, request: &DeleteReplicaSnapshot) -> Result<(), SvcError>;
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/io_engine/mod.rs]: Convention is `destroy`.
```suggestion
    async fn destroy_repl_snapshot(&self, request: &DestroyReplicaSnapshot) -> Result<(), SvcError>;
```

File: control-plane/stor-port/src/types/v0/transport/snapshot.rs [hunk 2]
```diff
@@ -99,3 +99,8 @@ impl GenericSnapshotParameters {
         &self.name
     }
 }
+
+/// The request type to delete a snapshot of a replica.
+pub struct DeleteReplicaSnapshot {
+    pub snap_id: SnapshotId,
```
Comment (tiagolobocastro) [inline control-plane/stor-port/src/types/v0/transport/snapshot.rs:106]: missing doc comment

File: control-plane/agents/src/bin/core/node/wrapper.rs [hunk 3]
```diff
@@ -1384,19 +1384,26 @@ impl NexusChildActionApi<NexusChildActionContextNode> for Arc<tokio::sync::RwLoc
...
 
 #[async_trait]
 impl ReplicaSnapshotApi for Arc<tokio::sync::RwLock<NodeWrapper>> {
     async fn create_repl_snapshot(
         &self,
-        _request: &CreateReplicaSnapshot,
+        request: &CreateReplicaSnapshot,
     ) -> Result<CreateReplicaSnapshotResp, SvcError> {
-        todo!()
+        let dataplane = self.grpc_client_locked(request.id()).await?;
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/node/wrapper.rs]: this will clash with my PR, merging mine now, please rebase and resolve.

**Review by tiagolobocastro — APPROVED** [review]

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/agents/src/bin/core/controller/io_engine/mod.rs (hunk 1 above)
Comment (dsharma-dc) [reply control-plane/agents/src/bin/core/controller/io_engine/mod.rs]: Modified. The gRPC type name will remain DeleteReplicaSnapshotRequest though.

File: control-plane/agents/src/bin/core/node/wrapper.rs (hunk 3 above)
Comment (dsharma-dc) [reply control-plane/agents/src/bin/core/node/wrapper.rs]: rebased and updated.

File: control-plane/agents/src/bin/core/controller/io_engine/mod.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/agents/src/bin/core/controller/io_engine/mod.rs]: Yeah, we need to rename the dataplane api.. for next time..

**Comment by dsharma-dc** [comment]
bors merge

**Pull request merged**